Moli/Core/Screens/Report/ReportView.swift

```diff
@@ -2,12 +2,51 @@ import SwiftUI
 
 struct ReportView: View {
     var user: API.User.UserMinified
+    var data = ReportViewModel()
+    @State var successfulReportSent = false
+
+    func handleReportPress(item: ReportItem) {
+        Task {
+            do {
+                try await API.User.reportUser(userId: user.userId, reason: item.reason, description: item.description)
+                self.successfulReportSent = true
+            } catch {
+                GlobalSnackbar.shared.showSnackbar(
+                    .init(text: "Something went wrong, could not report user", variant: .danger)
+                )
+            }
+        }
+    }
 
     var body: some View {
         VStack {
-            Text("Report")
+            if successfulReportSent {
+                ReportFinalView()
+            } else {
+                List {
+                    ForEach(data.reportData, id: \.reason) { item in
```

Comment:
:art: Istället för att bli tvingad att skriva ett `id` här, så kan du göra list objektet till en Identifiable, så att:

```swift
struct ReportItem: Identifiable {
    var reason: String
    var description: String
    var options: [ReportItemOption]?
    var id: String { reason }
}
```

```suggestion
                    ForEach(data.reportData) { item in
```